portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -56,127 +64,232 @@ public void tearDown() {
+    dataServelet.doPost(postRequest, postResponse);
     when(postRequest.getParameter("text-input")).thenReturn("Comment2");
-    new DataServlet().doPost(postRequest, postResponse);
+    dataServelet.doPost(postRequest, postResponse);
     when(postRequest.getParameter("text-input")).thenReturn("Comment3");
-    new DataServlet().doPost(postRequest, postResponse);
+    dataServelet.doPost(postRequest, postResponse);
     
-    Assert.assertEquals(3, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+    assertThat(3).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
   }
 
   @Test 
-  public void testPOSTNull() throws Exception {
-    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
-    HttpServletRequest postRequest = mock(HttpServletRequest.class);
-    HttpServletResponse postResponse = mock(HttpServletResponse.class);
-
-    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  public void testPostNull() throws Exception {
+    assertThat(0).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
 
     when(postRequest.getParameter("text-input")).thenReturn(null);
-    new DataServlet().doPost(postRequest, postResponse);
+    dataServelet.doPost(postRequest, postResponse);
     
-    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+    assertThat(0).isEqualTo(ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
   }
 
   @Test
-  public void testGETSingleComment() throws Exception {
-    HttpServletRequest postRequest = mock(HttpServletRequest.class);
-    HttpServletResponse postResponse = mock(HttpServletResponse.class);
-    HttpServletRequest getRequest = mock(HttpServletRequest.class);
-    HttpServletResponse getResponse = mock(HttpServletResponse.class);
-
-    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+  public void testGetSingleComment() throws Exception {
+    long TIMESTAMP_ONE = 0;
+    Entity comment = new Entity(COMMENT_TABLE_NAME);
+    comment.setProperty(COMMENT_COLUMN_NAME, "Test Comment #1");
+    comment.setProperty(TIMESTAMP_COLUMN_NAME, TIMESTAMP_ONE);
+    ds.put(comment);
 
     StringWriter stringWriter = new StringWriter();
     PrintWriter printWriter = new PrintWriter(stringWriter);
     when(getResponse.getWriter()).thenReturn(printWriter);
 
-    new DataServlet().doPost(postRequest, postResponse);
-    new DataServlet().doGet(getRequest, getResponse);
+    dataServelet.doPost(postRequest, postResponse);
+    dataServelet.doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo("Test Comment #1");
+  }
+  
+  @Test
+  public void testGetSomeComments() throws Exception {
+    int NUM_COMMENTS = 2;
```

Comment:
just a note that UPPER_CASE should be reserved for true constants at the class level. These var should be lower camel case (but I think my comment below will make them unnecessary)

Reply:
done!